runtime/plugins/org.jboss.tools.runtime.core/src/org/jboss/tools/runtime/core/extract/ExtractUtility.java

```diff
@@ -41,6 +46,20 @@ public ExtractUtility(File file) {
 			util = new UntarGZUtility(file);
 		}
 	}
+	public ExtractUtility(File file, int format) {
+		this.file = file;
+		switch(format) {
+			case FORMAT_ZIP:
+				util = new UnzipUtility(file);
+				break;
+			case FORMAT_TAR:
+				util = new UntarUtility(file);
+				break;
+			case FORMAT_TGZ:
+				util = new UntarGZUtility(file);
+				break;
+		}
```

Comment:
what if it's not supported? is there a default strategy? should it throw an exception?